front/src/statistics/Statistics.js

```diff
@@ -0,0 +1,310 @@
+    const newDateObject = new Date(newDate);
+    setDate(newDateObject);
+    continueWithDate();
+  };
+
+  if (monthlyUsers?.length > 0) {
+    const total = countMonthlyVisitors();
+
+    return (
+      <div className="container">
+        {/* Section for selecting date */}
+        <div className="firstSection">
+          <form onSubmit={continueWithDate}>
+            { /* Datepicker */}
+            <MuiPickersUtilsProvider
+              utils={MomentUtils}
+              locale={lang}
+            >
+              <KeyboardDatePicker
+                autoOk
+                margin="normal"
+                name="date"
+                label={statistics.DayChoose[fin]}
+                value={date}
+                onChange={(newDate) => handleDateChange(newDate)}
+                onAccept={handleDatePickChange}
+                format="DD.MM.YYYY"
+                showTodayButton
+              />
+            </MuiPickersUtilsProvider>
+          </form>
+        </div>
+        <hr />
+
+        {/* Header with arrows */}
+        <Grid class="date-header">
+          <div
+            className="hoverHand arrow-left"
+            onClick={previousDayClick}
+          />
+          <h1 className="headerText">
+            {date.toLocaleDateString('fi-FI')}
+          </h1>
+          <div
+            className="hoverHand arrow-right"
+            onClick={nextDayClick}
+          />
+        </Grid>
+        {/* Charts */}
+        <div className="row">
+          <div className="mixed-chart">
+            {/* Labels */}
+            <h2>{statistics.Day[fin]}</h2>
+            <h3>
+              {`${statistics.Total[fin]} ${date.toLocaleDateString('fi-FI')}: ${monthlyUsers[dayNumber - 1]}`}
+            </h3>
+            <div className="bar">
+              <Chart
+                options={dayOptions}
+                series={daySeries}
+                type="bar"
+                width="700"
+                height="400"
+              />
+            </div>
+            {/* Labels */}
+            <h2>{statistics.Month[fin]}</h2>
+            <h3>
+              {`${statistics.Total[fin]} ${(date.getMonth() + 1)}/${date.getFullYear()}: ${total}`}
+            </h3>
+            <div className="line">
+              <Chart
+                options={monthOptions}
+                series={monthSeries}
+                type="line"
+                width="700"
+                height="400"
+              />
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+  return (<div />);
+};
+
+async function getMonthlyVisitors(firstDate, lastDate) {
+  const query = `api/daterange/freeform/${firstDate}/${lastDate}`;
+  const response = await axios.get(query);
+  if (response) {
+    // Form an array including the visitors of a certain month and return it
+    let visitors = [];
+    response.data.forEach((supervision) => {
+      if (!supervision?.scheduleId) {
+        visitors = visitors.concat(0);
```

Comment:
Concat is used to concat arrays together. Use visitors.push(0) instead.

Reply:
I tried to use push() instead of concat() but it said visitors.push is not a function

Reply:
Interesting, could you post your code here?

Reply:
I tried this but it had multiple errors. The latter for-loop and the push-functions don't work.
```suggestion
async function getMonthlyVisitors(firstDate, lastDate) {
  const query = `api/daterange/freeform/${firstDate}/${lastDate}`;
  const response = await axios.get(query);
  if (response) {
    // Form an array including the visitors of a certain month and return it
    let visitors = [];
    for (var i = 0; i <= response.data.length; i++) {
      let supervision = response.data[i];
      if (!supervision?.scheduleId) {
        visitors = visitors.push(0);
      } else {
        let trackVisitors = 0;
        for (var j = 0; j <= supervision.tracks.length; j++) {
          let track = supervision.tracks[j];
          trackVisitors += track.scheduled.visitors;
        };
        visitors = visitors.push(trackVisitors);
      }
    };
    return visitors;
  }
  return [];
}
```

Reply:
push() is in-place, you don't need to reassign the variable again; that is:

visitors.push(0)

instead of

visitors = visitors.push(0)

For the second loop, you can loop through the values of the supervision.tracks by doing this:

for (const track of supervision.tracks) {
    trackVisitors += track.scheduled.visitors
}

Hopefully this'll solve the problems.

Reply:
As a matter of fact, you can use a similar loop in the first loop too

Reply:
Also instead of using

if (response) { ...

you can rather use

if (!response) {
    return [];
}

let visitors = 0;
...

This is a trick to reduce the amount of indentations and code paths.

Reply:
thanks!